06-building-a-program-start-to-finish/writing-tests/Output/inverted-index-spec.js

```diff
@@ -0,0 +1,180 @@
+/* global describe, it, expect */
+const fantasyBook = require('../../src/test books/fantasyBook.json'),
+      quotesAndRap = require('../../src/test books/quotesAndRap.json'),
+      religiousBook = require('../../src/test books/religiousBook.json'),
+      testsBook = require('../../src/test books/testsBook.json'),
+      notArray = require('../../src/test books/notArray.json'),
+      notArrayOfObjects =
+                      require('../../src/test books/notArrayOfObjects.json'),
+      moreThanTwoKeys = require('../../src/test books/moreThanTwoKeys.json'),
+      notAllTextKeys = require('../../src/test books/notAllTextKeys.json'),
+      capitalTitleKey= require('../../src/test books/capitalTitleKey.json'),
+      titleValueAsNumber =
+                        require('../../src/test books/titleValueAsNumber.json'),
+      textValueAsNumber =
+                        require('../../src/test books/textValueAsNumber.json');
+      religiousBookIndexes =
+                      require('../../src/test books/religiousBookIndexes.json');
+
+
+const invertedIndex = new InvertedIndex();
+
+describe('InvertedIndex', () => {
+  
+  describe('constructor', () => {
+    it('should be an instance of  InvertedIndex', () => {
+      expect(invertedIndex instanceof InvertedIndex).toBeTruthy();
+      expect(typeof (invertedIndex)).toEqual('object');
+    });
+
+    it('should have defualt instatiated  values', () => {
+      expect(invertedIndex.allIndexed).not.toBe(null);
+      expect(invertedIndex.allTitles).not.toBe(null);
+      expect(invertedIndex.documentsIndex).not.toBe(null);
+      expect(invertedIndex.allIndexed).toEqual({});
+      expect(invertedIndex.allTitles).toEqual({});
+      expect(invertedIndex.documentsIndex).toEqual({});
+      expect(invertedIndex.allIndexed).toBeDefined();
+      expect(invertedIndex.allTitles).toBeDefined();
+      expect(invertedIndex.documentsIndex).toBeDefined();
+    });
+  });
+
+  describe('readFile', () => {
+    it('should be defined', () => {
+      expect(invertedIndex.readFile).not.toBeUndefined();
+    });
+
+    it('should be a function', () => {
+      expect(typeof invertedIndex.readFile).toBe('function');      
+    });
+  });
+
+  describe('tokenize', () => {
+    it('should eliminate special characters, symbols and white spaces', () => {
+      const textString = 'why are We\n so blind\t to see two-timers#@!`~_+=/?"';
+
+      const tokenizedTextString = [ 'why', 'are', 'we', 'so', 'blind', 'to',
+                                    'see', 'two', 'timers' ];
+                                    
+      expect(invertedIndex.tokenize).not.toBeUndefined();
+      expect(typeof invertedIndex.tokenize).toBe('function');
```

Comment:
I see you doing this on every test suite for different methods
`expect(typeof invertedIndex.methodName).toBe('function');`

I feel that it would be better to have them all on a separate Test Suite